cmd/shipper/main.go

```diff
@@ -510,13 +513,39 @@ func startTrafficController(cfg *cfg) (bool, error) {
 	return true, nil
 }
 
+func startRolloutBlockController(cfg *cfg) (bool, error) {
+	enabled := cfg.enabledControllers["rolloutblock"]
+	if !enabled {
+		return false, nil
+	}
+
+	c := rolloutblock.NewController(
+		buildShipperClient(cfg.restCfg, rolloutblock.AgentName, cfg.restTimeout),
+		cfg.shipperInformerFactory,
+		cfg.recorder(rolloutblock.AgentName),
+	)
+
+	cfg.wg.Add(1)
+	go func() {
+		c.Run(cfg.workers, cfg.stopCh)
+		cfg.wg.Done()
+	}()
+
+	return true, nil
+}
+
 func startWebhook(cfg *cfg) (bool, error) {
 	enabled := cfg.enabledControllers["webhook"]
 	if !enabled {
 		return false, nil
 	}
 
-	c := webhook.NewWebhook(cfg.webhookBindAddr, cfg.webhookBindPort, cfg.webhookKeyPath, cfg.webhookCertPath)
+	c := webhook.NewWebhook(
+		cfg.webhookBindAddr,
+		cfg.webhookBindPort,
+		cfg.webhookKeyPath,
+		cfg.webhookCertPath,
+		buildShipperClient(cfg.restCfg, rolloutblock.AgentName, cfg.restTimeout))
```

Comment:
Let's move `buildShipperClient` outside of this call. Looking at the argument list (say, the timeout value), it makes me think the building can fail eventually and we might consider checking err-nilness before creating a webhook

Reply:
And incase `buildShipperClient` returns a nil object, should we raise an error?

Reply:
`buildShipperClient` (and its friend buildKubeClient) actually panic on error. Unless we plan on changing that (and I think that should probably go on a different PR), the call here is appropriate.